response.go

```diff
@@ -127,6 +127,23 @@ func MarshalManyPayload(w io.Writer, models interface{}) error {
 	return nil
 }
 
+func MarshalManyPayloadWithLinks(w io.Writer, models interface{}, links map[string]interface{}) error {
```

Comment:
@sbelkin88 so that we do not change the surface area of the public api, please add an interface that has a method like `Links() map[string]interface{}`, that you can type assert.  Iff the model implements the interface, serialize the links.